File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        obj = {}
        for i in nums:
            if i in obj:
                obj[i] += 1
            else:
                obj[i] = 1
        for key, value in obj.items():
            if value > len(nums) / 2:
                return key

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map <int,int> count={};
        int n=nums.size()/2;
        for(int i : nums){
            if(count[i]){
                count[i]=count[i]+1;
            }
            else{
                count[i]=1;
            }
            if(count[i]>n){
                return i;
            }
        }
        return 0;
    }
};